src/cljs/pancetta/components/home.cljs

```diff
@@ -1,23 +1,86 @@
 (ns pancetta.components.home
   (:require [matchbox.core :as m]
+            [pancetta.common.helpers :as h :refer [price-to-str]]
+            [secretary.core :as secretary]
             [reagent.core :as reagent :refer [atom]]))
 
-(defn home-component [state]
+(def style {:form-container {:background-color "white"
+                             :padding "20px 30px"
+                             :margin 5}
+            :results-container {:margin 5}
+            :ticket {:background-color "white"
+                     :margin 1
+                     :border-radius 3
+                     :padding 10
+                     :cursor "pointer"}
+            :form-group {:margin "20px 0px"}})
+
+(def query (atom {:from "London"
+                  :to "Paris"
+                  :on (-> (js/Date.) .toISOString (.slice 0 10))}))
+
+(def tickets (atom nil))
+
+;; should be retrieved from firebase
+(def cities ["Paris", "London", "Bruxelles"])
+
+(defn get-tickets [root]
   (let [tickets-ref (->
-                      (m/get-in (:root @state) [:tickets])
-                      (.orderByChild "price/amount")
-                      (.limitToLast 10))
-        tickets (atom nil)]
-    (m/listen-to tickets-ref :value #(reset! tickets (get % 1)))
-    (fn []
-      [:div
-        [:h1 {:on-click #(.log js/console "click")} "HOME PAGE"]
-        [:div
-          (for [[ticket-id ticket] @tickets]
-            [:div {:key ticket-id}
-              (:from ticket)
-              " -> "
-              (:to ticket)
-              " on: " (:on ticket)
-              " for "
-              (get-in ticket [:price :amount])])]])))
+                     (m/get-in root [:tickets])
+                     (.orderByChild "on")
+                     (.equalTo (:on @query)))]
+    (m/listen-to tickets-ref
+                 :value
+                 (fn [[id res]] (reset! tickets
+                                        (doall (filter
+                                                (fn [[id ticket]] (and (= (:to @query) (:to ticket))
+                                                                       (= (:from @query) (:from ticket)))) res)))))))
+
+(defn ticket-component [id ticket]
+  [:div {:style (:ticket style)
+         :on-click #(secretary/dispatch! (str "ticket/" (name id)))}
+   [:span (str "Ticket from " (:from ticket))]
+   [:span (str " to " (:to ticket))]
+   [:span (str " for " (price-to-str (:price ticket)))]])
+
+(defn form-component [state]
+  [:div {:class-name "pure-u-3-5"}
+    [:form {:style (:form-container style) :class-name "pure-form"}
+     [:h2 "Find a ticket"]
+     [:div {:style (:form-group style)}
+      [:span "I am looking for a train on the "]
+      [:input {:type "date"
+               :min (-> (js/Date.) .toISOString (.slice 0 10))
```

Comment:
I didn't know you created a component for that. It could be a common component yes!